### PR TITLE
Feat(): pass return url (new payment methods)

### PR DESCRIPTION
**Pull request by grzesebe**
Hi,
I was investigating possibility of using different payment methods ex: p24, Klarna...
It occurs that we only need to update the SDKs
On IOS it's also needed to pass "returnURL" param and handle URL callback after return to app from browser. 
https://stripe.com/docs/payments/accept-a-payment?platform=ios&ui=payment-sheet#ios-set-up-return-url 

I've tested it on IOS and Android Emulator and it seems to work pretty well.

Unfortunately I wasn't able to compile on Android without bumping `targetSdkVersion` and `compileSdkVersion`

I'm handling URL like this:
```javascript
import { App as CapacitorApp } from '@capacitor/app';
import { Stripe } from '@capacitor-community/stripe';

function initUrlHandler(router) {
  CapacitorApp.addListener('appUrlOpen', function (event) {
    console.log('app opened from url', event);
    try {
      const slug = event.url.split('://').pop();
      if (/same-page((\?|#).*)?$/gm.test(slug)) return;
      if (/stripe-callback((\?|#).*)?$/gm.test(slug)) {
        Stripe.handleURLCallback({ url: event.url });
        return;
      }
      const resolved = router.resolve(slug);
      if (slug && (!resolved || resolved.name === '404')) {
        throw new Error('Nieznana ścieżka aplikacji');
      }
      if (slug) router.push(resolved);
    } catch (err) {
      alert(err);
    }
  });
}

export default initUrlHandler;
```

### CODE REVIEW

**Comment by grzesebe** [comment]
I've tested this solution with p24 (https://stripe.com/docs/payments/p24), and 3DSecure card confirmation (https://stripe.com/docs/payments/3d-secure)

File: android/build.gradle [hunk 1]
```diff
@@ -22,10 +22,10 @@ apply plugin: 'kotlin-android'
 apply plugin: 'kotlin-android-extensions'
 
 android {
-    compileSdkVersion project.hasProperty('compileSdkVersion') ? rootProject.ext.compileSdkVersion : 30
+    compileSdkVersion project.hasProperty('compileSdkVersion') ? rootProject.ext.compileSdkVersion : 32
```
Comment (rdlabo) [inline android/build.gradle:25]: Capacitor 3's target & compile sdkVersion is 30. So if you need it, you will have to wait for a Capacitor4-compatible version.

File: ios/Plugin/StripePlugin.swift [hunk 2]
```diff
@@ -34,6 +34,32 @@ public class StripePlugin: CAPPlugin {
...
+        }
+
+        let url = URL(string: urlString)!
+        DispatchQueue.main.async {
+            let stripeHandled = StripeAPI.handleURLCallback(with: url)
+            if (stripeHandled) {
+                call.resolve()
+            } else {
+                call.reject("This was not a Stripe url – handle the URL normally as you would")
+                return
```
Comment (rdlabo) [inline ios/Plugin/StripePlugin.swift:56]: This may not need

**Comment by rdlabo** [comment]
@grzesebe Thanks for your great work and pull request! 
I was in a bit of a hurry so I couldn't take a closer look, but I commented. Please let me know if it is mandatory, especially for Android targetVersion.

File: android/build.gradle (hunk 1 above)
Comment (grzesebe) [reply android/build.gradle:25]: The new payment methods was added in v19.1.0 of Android Stripe SDK (https://github.com/stripe/stripe-android/pull/4489).
Unfortunately compileSdkVersion was bumped in 18.2.0 (https://github.com/stripe/stripe-android/commit/705390657782b2ce820304e92a536105c774db12).

Maybe there is a way to choose a SKD version based on compileSdkVersion of parent project?

**Comment by grzesebe** [comment]
@rdlabo 
Thanks for your hard work on this plugin. You're the best.
The p24 payment is crucial for me before using this library in production. The card payments are not enough.
Maybe I should create a temporary NPM package until we are able to update the android SDK version?

**Comment by rdlabo** [comment]
In fact, we have already pre-released a version that supports Capacitor4, and the android SDK version is already available in the master branch.

Now you no longer have a problem accepting your pull request. Now we'll work on resolving the conflicts and doing the merge!

**Comment by rdlabo** [comment]
Merged!!

@grzesebe Thanks for great work!!
Did you use this plugin for production? If yes, please tell us what app. This help development this plugin.

👉 https://github.com/capacitor-community/stripe/issues/145

Thanks.

**Comment by grzesebe** [comment]
Wow, that was fast :)
As soon as we will go production I'll post in https://github.com/capacitor-community/stripe/issues/145

**Pull request merged**